### ISSUE TITLE
An expression makes RuleSet.DNF looping forever

### ISSUE THREAD

**Issue by Marco91460** [bpodgursky/jbool_expressions#43]
Hi,

Following expression : 
String target ="((A34 | A35 | A36 | A37 | A38 | A39) & (A30 | A31) & (A32 | A33) & ((A27 & (A20 & A21 & A22)) | (A28 & (A23 & A24 & (A22 | A25))) | (A29 & (A23 & A26 & (A22 | A25)))) & (A40 | A41 | A42 | A43 | A44) & (((A20 & A21 & A22) | (A22 & A23 & A24) | (A23 & A24 & A25) | (A23 & A26 & (A22 | A25))) & A17 & (A18 | A19)) & (A1 & A2 & (((A3 | A4 | A5 | A6 | A7 | A8 | A9 | A10 | A11 | A12) & A13 & A14) | (A15 & A16))))"

makes RuleSet.toDNF never returning a result.

		Expression<String> nonStandard = null;
		try {
			nonStandard = RuleSet.simplify(ExprParser.parse(target));
		} catch (Exception e) {
			throw new BadDNFException("Error in following "+target);
		}
		System.out.println("Result Simplified = "+nonStandard.toString());
		System.out.flush();
		Expression<String> dnfForm = RuleSet.toDNF(nonStandard);

Thanks

**Comment by zaza** [comment]
I can confirm this is still an issue. It's been over 10 minutes and the snippet is still running.

**Comment by bpodgursky** [comment]
Do you know if the DNF version of this is actually reasonably compact?  Sometimes you end up with enormously large DNF expressions when you multiply it all out and there's not much to be done about it.

**Comment by zaza** [comment]
I doubt it is. A subexpression from @Marco91460 's example is 74399 characters long (`toString().length()`):

`((A34 | A35 | A36 | A37 | A38 | A39) & (A30 | A31) & (A32 | A33) & ((A27 & (A20 & A21 & A22)) | (A28 & (A23 & A24 & (A22 | A25))) | (A29 & (A23 & A26 & (A22 | A25)))) & (A40 | A41 | A42 | A43 | A44) & (((A20 & A21 & A22) | (A22 & A23 & A24) | (A23 & A24 & A25) | (A23 & A26 & (A22 | A25))) & A17 & (A18 | A19)) `

**Comment by bpodgursky** [comment]
Yeah... I'm sure there are some computational tricks we could use to spit out a gig of DNF text faster, but I don't think the output is realistically going to be useful.  I'm kind of inclined to close this out unless anyone has a concrete suggestion.
